python/ray/tune/trial_scheduler.py

```diff
@@ -64,6 +72,14 @@ def debug_string(self):
 class FIFOScheduler(TrialScheduler):
     """Simple scheduler that just runs trials in submission order."""
 
+    def track_experiment(self, experiment, trial_runner):
+        generator = generate_trials(experiment.spec, experiment.name)
+        while True:
+            try:
+                trial_runner.add_trial(next(generator))
+            except StopIteration:
+                break
```

Comment:
Move this into TrialScheduler?

Reply:
no strong opinion about this; is there a benefit to keeping TrialScheduler as an interface?

Reply:
It doesn't seem particularly specific to fifo scheduler. I imagine most schedulers will want to extend this instead of fifoscheduler.

Speaking of which should HyperOpt extend the scheduler interface instead of fifoscheduler?

Reply:
I think wrt extending HyperOpt it's fine to extend fifo, as it does call `FIFOScheduler` methods for `on_trial_add` and also for `choose_trial_to_run`. 

That being said, the `choose_trial_to_run` invocation is done as a weird class method invocation rather than a super invocation and could work either way, so I'm not married to anything...